### PR TITLE
Refs task #373743 Add webhook

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#373743](https://dev.sun-asterisk.com/issues/373743)

## What's this PR do ?

- [x] Coding for add add new webhook

## Library
*(List package, library third party add new include version)*

- laravelcollective/html : ^5.8.0
- sun-asterisk/chatwork-php: ^0.1.2

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*
image

### CODE REVIEW

File: app/Http/Controllers/WebhookController.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(WebhookCreateRequest $request)
+    {
+        $data = $request->except('_token');
+        $data['token'] = ''.Auth::id().''.substr(str_shuffle(config('settings.letters')), 0, 30).''.time();
```
Comment (phucqd) [inline app/Http/Controllers/WebhookController.php]: cái $data[' token'] này để làm gì thế em nhỉ

Comment (ghost) [reply app/Http/Controllers/WebhookController.php]: cái này tạo e random 1 chuỗi string cho trường token của webhook ở [đây](https://github.com/sun-asterisk-research/chatwork-forwarder/blob/develop/database/migrations/2019_11_19_060449_create_webhooks_table.php#L22) á anh

Comment (ghost) [reply app/Http/Controllers/WebhookController.php]: e update lại dùng md5(user_id+current_timestamp) rồi nhé a

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>failure</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/RoomController.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+class RoomController extends Controller
+{
+    /**
+     * Return a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $bot = Bot::findOrFail($request->bot_id);
```
Comment (tamtt-1031) [inline app/Http/Controllers/RoomController.php:19]: trường hợp bot chưa tìm thấy xử lý chưa nhỉ

Comment (ghost) [reply app/Http/Controllers/RoomController.php:19]: findOrFail khi không tìm thấy sẽ bắn ra execption và trả về 404 response, e bắt bên js ở [đây](https://github.com/sun-asterisk-research/chatwork-forwarder/pull/11/files#diff-db134f314bbd12cfca2a37a17429566fR17-R21) rồi ạ

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>failure</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>failure</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>failure</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr></table>

**Review by nguyenthinhhn — APPROVED** [review]
ok

**Review by tamtt-1031 — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**